pandas/core/groupby.py

```diff
@@ -2368,6 +2368,13 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     # axis of the object
     if level is not None:
         if not isinstance(group_axis, MultiIndex):
+            if isinstance(level, list) or isinstance(level, range):
```

Comment:
Changed to `is_list_like`. `range` objects still work, since they are iterables (so they result in `True` under `is_list_like`) and the builtin `set` accepts them.